project1/src/Pages/Profile.tsx

```diff
@@ -24,6 +28,19 @@ const Profile: React.FC<RouteComponentProps<{ user: string }>> = props => {
   });
 
   useEffect(() => {
+    if (profile.username) {
```

Comment:
can please you add comments these conditions? So we know what is hapening
and shouldn't you split these in two `useEffect`? seperation of concerns